okhttp-ws/src/main/java/com/squareup/okhttp/ws/WebSocketCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.okhttp.ws;
+
+import com.squareup.okhttp.Response;
+import java.io.IOException;
+
+public interface WebSocketCallback {
+  /**
+   * Called when the request has successfully been upgraded to a web socket. You <b>must</b> call
+   * {@link WebSocket#start(WebSocket.Listener) start()} on the {@code WebSocket} with a listener.
```

Comment:
This interface feels labor intensive. Let's do a Hangout & you can better explain to me what the moving parts are!